docs/paper/dev/api/entity-api/entity-teleport.mdx

```diff
@@ -52,7 +59,7 @@ All available teleport flags can be found in the <Javadoc name={"io.papermc.pape
 Teleport a player relatively, preventing velocity from being reset in the X, Y and Z axes.
 
 ```java
-player.teleport(
+player.teleportAsync(
```

Comment:
afaik, `teleportAsync` also has to be run on the main thread, it just won't block the main thread while loading chunks for the location.

Reply:
I looked into the patch and didn't see a catcher, will add a warning box though

Reply:
I could be wrong about this, cause now it looks like the getChunkAtAsync method (which is what teleportAsync uses) does have separate logic if called off the main thread.

Reply:
tried running `teleportAsync` in `ForkJoinPool#commonPool` and joined the future, worked just fine

`teleport` failed as expected:
```
[19:10:15 WARN]: Exception in thread "ForkJoinPool.commonPool-worker-1" java.lang.IllegalStateException: PlayerTeleportEvent may only be triggered synchronously.
[19:10:15 WARN]: 	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:42)
[19:10:15 WARN]: 	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:131)
[19:10:15 WARN]: 	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:627)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftPlayer.teleport(CraftPlayer.java:1450)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftPlayer.teleport(CraftPlayer.java:1348)
[19:10:15 WARN]: 	at org.bukkit.craftbukkit.entity.CraftEntity.teleport(CraftEntity.java:226)
[19:10:15 WARN]: 	at test-plugin-1.0.0-SNAPSHOT.jar//me.kcra.testplugin.TestPlugin.lambda$onEnable$9(TestPlugin.java:183)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1423)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
[19:10:15 WARN]: 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
```